### ISSUE TITLE
[shelly] ix3 changing to ON and OFF every 10s without interaction

### ISSUE THREAD

**Issue by TheTrueRandom** [openhab/openhab-addons#10847]
For my home setup I use shelly ix3 to control the lights.

## Expected Behavior
When the Shelly Ix3 is not touched, no event should be processed by openhab.

## Current Behavior
About every 10 seconds the lights turn on, stay on for a few seconds and turn off again. This is caused by the shelly binding sending updates to the channel where no change actually happened. 

Thing configuration:
`Thing shelly:shellyix3:bedroom              "Shelly i3 Light Bedroom" @ "Bedroom"                   [deviceIp="192.168.0.7", userId="", password=""]`

Item configuration:
`Switch                      ShellyI3LightBedroomStatus1Input       "Input"                                                     {channel="shelly:shellyix3:bedroom:status1#input"}`

I use this setup for 5 other rooms which works without the bug, so I'm not sure what causes it for this room. I checked with the shelly API and there is absolutely no change or event triggered. Also tried with specify an action URL on the shelly UI, this also works perfectly, only triggered when I physically press the button. 

Debug logs from openhab:
 ```
2021-06-11 02:05:52.920 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: CoIoT Message from /192.168.0.79:5683 (MID=59927): {"G":[[0,9103,3],[0,2101,0],[0,2102,""],[0,2103,12],[0,2201,0],[0,2202,""],[0,2203,50],[0,2301,0],[0,2302,""],[0,2303,0]]}
2021-06-11 02:05:52.921 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: CoIoT Sensor data {"G":[[0,9103,3],[0,2101,0],[0,2102,""],[0,2103,12],[0,2201,0],[0,2202,""],[0,2203,50],[0,2301,0],[0,2302,""],[0,2303,0]]} (serial=34560)
2021-06-11 02:05:52.922 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: 10 CoAP sensor updates received
2021-06-11 02:05:52.924 [DEBUG] [lly.internal.util.ShellyChannelCache] - shellyix3-bedroom: Channel status1#input updated with OFF (type class org.openhab.core.library.types.OnOffType).
2021-06-11 02:05:52.927 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: 1 channels updated from CoIoT status, serial=34560
```
```
2021-06-11 02:05:53.758 [DEBUG] [lly.internal.util.ShellyChannelCache] - shellyix3-bedroom: Channel status1#input updated with ON (type class org.openhab.core.library.types.OnOffType).
==> /var/log/openhab/events.log <==
2021-06-11 02:05:52.925 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ShellyI3LightBedroomStatus1Input' changed from ON to OFF
2021-06-11 02:05:53.760 [INFO ] [openhab.event.ItemStateChangedEvent ] - Item 'ShellyI3LightBedroomStatus1Input' changed from OFF to ON
==> /var/log/openhab/openhab.log <==
2021-06-11 02:05:54.045 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: CoIoT Message from /192.168.0.7:5683 (MID=28146): {"G":[[0,9103,0],[0,2101,1],[0,2102,""],[0,2103,1],[0,2201,0],[0,2202,""],[0,2203,0],[0,2301,0],[0,2302,""],[0,2303,0]]}
2021-06-11 02:05:54.046 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: CoIoT Sensor data {"G":[[0,9103,0],[0,2101,1],[0,2102,""],[0,2103,1],[0,2201,0],[0,2202,""],[0,2203,0],[0,2301,0],[0,2302,""],[0,2303,0]]} (serial=768)
2021-06-11 02:05:54.047 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: 10 CoAP sensor updates received
```

These log messages repeat every 10 seconds, also the serial is sometimes "34560" and sometimes "768", alternating between these values. For my other shellyix3 I can see in the logs the serial has only one value, not alternating. 

## Possible Solution
I think the duplicate handling is not catching this issue: https://github.com/openhab/openhab-addons/blob/a2084e746a45020814a048fcca6740909571185e/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/coap/ShellyCoapHandler.java#L270 
but I'm not entirely sure if that's the issue and also I don't know why it is only occurring for one shelly.

## Steps to Reproduce (for Bugs)
unfortunately I can't reproduce it for other shellys.

## Your Environment
Shelly firmware is the latest and same on all shellys:  20210429-102402/v1.10.4-g3f94cd7
openhab 3.0.2 running on raspberry pi 4, 3.0.2

**Comment by markus7017** [comment]
I see no commands in the log, just status updates from the device. Do ypu have configured an Auto On/Off timer in the device? or are you running some kind of rule?

**Comment by TheTrueRandom** [comment]
No to both. The status updates come from the channel, when I remove the channel binding of the item, the changes to the item are gone.

**Comment by TheTrueRandom** [comment]
Ok, I'm one step further analyzing. 
I removed the power supply to the device, in the log I can see the device status changes from offline to online and back frequently (even though no power is connected).
In the trace logs I can see the following:

```
2021-06-17 20:54:56.940 [DEBUG] [elly.internal.coap.ShellyCoapHandler] - shellyix3-bedroom: CoIoT Message from /192.168.0.79:5683 (MID=6717): {"G":[[0,9103,0],[0,2101,0],[0,2102,""],[0,2103,1],[0,2201,1],[0,2202,""],[0,2203,30],[0,2301,0],[0,2302,""],[0,2303,0]]}
```

how can this happen? The device with the IP 192.168.0.79 updates the bedroom device?

For reference: full thing configuration:
```
Thing shelly:shellyix3:bedroom              "Shelly i3 Light Bedroom" @ "Bedroom"                   [deviceIp="192.168.0.7", userId="", password=""]   
Thing shelly:shellyix3:corridor             "Shelly i3 Light Corridor" @ "Corridor"                 [deviceIp="192.168.0.88", userId="", password=""] 
Thing shelly:shellyix3:bathroom             "Shelly i3 Light Bathroom" @ "Bathroom"                 [deviceIp="192.168.0.200", userId="", password=""]  
Thing shelly:shellyix3:office               "Shelly i3 Office" @ "Office"                           [deviceIp="192.168.0.79", userId="", password=""]  
Thing shelly:shellyix3:livingroom           "Shelly i3 LivingRoom" @ "LivingRoom"                   [deviceIp="192.168.0.239", userId="", password=""] 
```

I tried to uninstall the binding, restart openhab and install the binding again which doesn't solve the issue. Somehow the deviceIp  from a different shelly is used to update another shelly.

**Comment by lapicidae** [comment]
I have the exact same problem with my shelly1pm.

**A possible workaround is:**
Disable "Auto-CoIoT" in the binding config and disable "CoIoT Events" in the thing config.

The device reacts slowly to changes, but there are no more ghostly "on / off" events.

P.S.: CoIoT mode/address was set to Multicast and IP... for testing.

**Comment by markus7017** [comment]
Does the problem still happen with latest DEV build?

**Comment by lapicidae** [comment]
> Does the problem still happen with latest DEV build?

Yes, nothing has changed with the last "docker:snapshot". (shelly 1PM)

[debugLog.txt](https://github.com/openhab/openhab-addons/files/8429204/debugLog.txt)

**Comment by markus7017** [comment]
Please try again

**Comment by markus7017** [comment]
Found it, stupid bug: The incoming Coop handler checks the ip source of the packet agains the thing.deviceIp to find the matching thing, but I used contains(), which means that a packet from your Bedroom ix3 (192.168.0.7) updates the Office ix3 (192.168.0.79), because "192.168.0.7" is also included in "192.168.0.79". Very strange that this was never reported before :-)

Try the updated DEV build

---

[3.4.3-DEV Gen1/Plus/Pro](https://github.com/markus7017/myfiles/blob/master/shelly/org.openhab.binding.shelly-3.4.3-SNAPSHOT.jar?raw=true) | [4.0.0-DEV Gen1/Plus/Pro](https://github.com/markus7017/myfiles/blob/master/shelly/org.openhab.binding.shelly-4.0.0-SNAPSHOT.jar?raw=true) | [README](https://github.com/markus7017/myfiles/blob/master/shelly/README.md) | **[READMEbeta](https://github.com/markus7017/myfiles/blob/master/shelly/READMEbeta.md)** 
[Avdanced Users](https://github.com/markus7017/myfiles/blob/master/shelly/doc/AdvancedUsers.md) - [Shelly Manager](https://github.com/markus7017/myfiles/blob/master/shelly/doc/ShellyManager.md) - [Bugs/Features](https://github.com/openhab/openhab-addons/issues?q=is%3Aissue+is%3Aopen+%5Bshelly%5D) -  [API Doc](https://shelly-api-docs.shelly.cloud/?fbclid=IwAR23ukCi_3aBSTPRHYUIcpr0pLi0vcyL0fF0PnJQdFvkkc8_Zo5LkAcli_A#http-server) | [Firmware Index](https://repo.shelly.cloud/files/firmware/?fbclid=IwAR0Kz6bjMGoq_HYU85QAdl8b-xh7E3IPa4ENKjqsvmMAYQ6QGC8ODvqPzUs) - [Firmware Archive](http://archive.shelly-faq.de)

---

Note: The DEV build is always newer than the version in the official Distro or Milestone builds

**Comment by lapicidae** [comment]
In my environment, the problem is not solved.

Device IP of the problematic device is "192.168.0.22"
It doesn't seem to be the problem you described above.

I used the bundle [org.openhab.binding.shelly-4.0.0-SNAPSHOT.jar](https://raw.githubusercontent.com/markus7017/myfiles/master/shelly/org.openhab.binding.shelly-4.0.0-SNAPSHOT.jar) from [https://github.com/markus7017/myfiles/tree/master/shelly](https://github.com/markus7017/myfiles/tree/master/shelly) and the latest "openhab/openhab:snapshot" docker image.
```
bundle:list | grep -i shelly
246 │ Active │  80 │ 4.0.0.202304021248     │ openHAB Add-ons :: Bundles :: Shelly Binding Gen1+2
```

[openhab.log](https://github.com/openhab/openhab-addons/files/11148572/openhab.log)
[events.log](https://github.com/openhab/openhab-addons/files/11148569/events.log)
